FileSets: if HB alerting on an invalid type, tag alert as data error

jmartin-sul opened the pull request:
## Why was this change made?

closes #3232

after grepping around to double-check that i was using the right alert format, i realized i'd missed some applicable centralized code for this sort of alert, but also that the existing code wanted a druid, which wasn't immediately available from where the alert was being sent.  i threw on a second commit passing in the druid from the caller, and switching `FileSets` to use the centralized notifier code, but if it turns out that's bad for some reason, i can knock that commit off (or otherwise tweak as needed, of course).

## How was this change tested?

unit tests

## Which documentation and/or configurations were updated?

n/a

justinlittman commented on app/services/cocina/from_fedora/file_sets.rb:
```diff
@@ -4,20 +4,22 @@ module Cocina
         @rights_metadata = rights_metadata
         @version = version
         @ignore_resource_type_errors = ignore_resource_type_errors
+        @data_err_notifier = DataErrorNotifier.new(druid: druid)
```
This should be passed as a parameter rather than instantiated.

justinlittman replied:
You can pass from Cocina::FromFedora::DRO: https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/from_fedora/dro.rb#L64

jmartin-sul commented:
ok, i switched to passing along the existing notifier instance, instead of instantiating anew based on a druid given to `FileSets.build`.  that resulted in a bit more churn than i'd have expected, so left that as a separate commit just in case we want to walk that back.  but since it sounds like it's fine to use the centralized notifier one way or the other, i squashed that change into the first commit, where the `data_error` tag was given to an inline `Honeybadger.notify` call.

i tested the validation script change by running `bin/validate-rights-cocina-roundtrip -s 100000` on sdr-deploy using this branch.  that completed with no problem, but it also didn't detect any errors mapping to either cocina or fedora, so i'm not sure that change got exercised.  but i was just following the pattern i saw in `bin/validate-cocina-roundtrip`, so hopefully that's ok.

i think this is ready for another look.
